sources/packages/backend/apps/api/src/services/application-bulk-withdrawal/application-bulk-withdrawal-import-business-validation.models.ts

```diff
@@ -0,0 +1,310 @@
+    },
+  )
+  @Length(SIN_NUMBER_LENGTH, SIN_NUMBER_LENGTH, {
+    message: `${DataTextHeaders.sin} must be of length ${SIN_NUMBER_LENGTH}.`,
+  })
+  sin: string;
+  /**
+   * Application Number.
+   */
+  @ValidateIf(
+    (object: ApplicationBulkWithdrawalImportBusinessValidationModel) =>
+      object.applicationFound,
+  )
+  @IsNumberString(
+    { no_symbols: true },
+    {
+      message: `${DataTextHeaders.applicationNumber} must be a valid number.`,
+    },
+  )
+  @Length(APPLICATION_NUMBER_LENGTH, APPLICATION_NUMBER_LENGTH, {
+    message: `${DataTextHeaders.applicationNumber} must be of length ${APPLICATION_NUMBER_LENGTH}.`,
+  })
+  applicationNumber: string;
+  /**
+   * Withdrawal Date.
+   */
+  @ValidateIf(
+    (object: ApplicationBulkWithdrawalImportBusinessValidationModel) =>
+      object.applicationFound,
+  )
+  @IsDate({
+    message: `${DataTextHeaders.withdrawalDate} must be a valid withdrawal date.`,
+  })
+  withdrawalDate: Date;
+  /**
+   * Application found.
+   */
+  @IsBoolean({
+    message: `${userFriendlyNames.applicationFound} must be a boolean value.`,
+  })
+  @IsIn([true], {
+    message: `${userFriendlyNames.application} is not present in SIMS and is a part of the SFAS system.`,
+    context: ValidationContext.CreateWarning(
+      ApplicationWithdrawalValidationWarnings.ApplicationNotFound,
+    ),
+  })
+  applicationFound: boolean;
+  /**
+   * Application belongs to institution.
+   */
+  @ValidateIf(
+    (object: ApplicationBulkWithdrawalImportBusinessValidationModel) =>
+      object.applicationFound,
+  )
+  @IsBoolean({
+    message: `${userFriendlyNames.applicationBelongsToInstitution}  must be a boolean value.`,
+  })
+  @IsIn([true], {
+    message: `${userFriendlyNames.application} does not belong to this institution.`,
+  })
+  applicationBelongsToInstitution: boolean;
+  /**
+   * Valid SIN.
+   */
+  @ValidateIf(
+    (object: ApplicationBulkWithdrawalImportBusinessValidationModel) =>
+      object.applicationFound,
+  )
+  @IsBoolean({
+    message: `${userFriendlyNames.validSIN} must be a boolean value.`,
+  })
+  validSIN: boolean;
+  /**
+   * Has correct institution code.
+   */
+  @ValidateIf(
+    (object: ApplicationBulkWithdrawalImportBusinessValidationModel) =>
+      object.applicationFound,
+  )
+  @IsBoolean({
+    message: `${userFriendlyNames.hasCorrectInstitutionCode} must be a boolean value.`,
+  })
+  @IsIn([true], {
+    message: `${userFriendlyNames.institutionCode} provided is incorrect.`,
+  })
+  hasCorrectInstitutionCode: boolean;
+  /**
+   * Application Status.
+   */
+  @ValidateIf(
+    (object: ApplicationBulkWithdrawalImportBusinessValidationModel) =>
+      object.applicationFound,
+  )
+  @IsEnum(ApplicationStatus)
+  @IsIn([ApplicationStatus.Completed], {
+    message: `${userFriendlyNames.application} is not in the ${ApplicationStatus.Completed} status.`,
+  })
+  applicationStatus: ApplicationStatus;
```

Comment:
If the application is not found many properties will be potentially not defined, rght?
The properties in the model must express it.